Best practices/React/Testing.md

```diff
@@ -84,7 +89,7 @@ There are many parts of React components that should not be tested as they are i
 
 * `state`. State is not directly manipulable from other components, and so should be treated as private. In order to simulate setting state, trigger the props of components rendered by your component with arguments that result in the component getting into that state. Once a component is in a state, do not assert on the state having a particular shape; instead, assert that the subcomponents being rendered have the expected props.
 
-* Instances for class components. All methods should be private on your React components, as you will otherwise be encouraging users to break out of the declarative model of React. This includes React’s lifecycle methods, which are implementation details of the framework itself. Never call any of these methods directly.
+* Instances for class components (or of function components, through `React.useImperativeHandle`). All methods should be private on your React components, as you will otherwise be encouraging users to break out of the declarative model of React. This includes React’s lifecycle methods, which are implementation details of the framework itself. Never call any of these methods directly.
```

Comment:
I feel like this is a bit mixed in it's messaging since `useImperativeHandle` is explicitly for defining these types of methods, whereas class methods appearing on the instance is a natural result of programming using classes.

I'm not sure there's ever a reason to call `useImperativeHandle` and pass in an object with private methods, so I'd prefer a blanket recommendation against using it.